File: bun_hono/src/controllers/healthController.ts
import type { Context } from 'hono';
import { localizedMessage, parseAcceptLanguage } from '../models/locale';

export const healthCheck = (c: Context) => {
  const language = parseAcceptLanguage(c.req.header('accept-language'));
  return c.json({ message: localizedMessage(language, 'success', 'thành công') });
};

File: back-end/src/controllers/health_controller.rs
use axum::Json;
use serde::Serialize;

use crate::locale::Language;

#[derive(Serialize, utoipa::ToSchema)]
pub struct HealthResponse {
    pub message: String,
}

#[utoipa::path(
    get,
    path = "/health",
    tag = "health",
    responses((status = 200, body = HealthResponse))
)]
pub async fn health_check(language: Language) -> Json<HealthResponse> {
    Json(HealthResponse {
        message: language.message("success", "thành công"),
    })
}
